pkgs/development/libraries/boost/cross.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/bootstrap.sh b/bootstrap.sh
+index ca0b08d58..5d4105ffe 100755
+--- a/bootstrap.sh
++++ b/bootstrap.sh
+@@ -223,7 +223,7 @@ rm -f config.log
+ if test "x$BJAM" = x; then
+   $ECHO -n "Building Boost.Build engine with toolset $TOOLSET... "
+   pwd=`pwd`
+-  (cd "$my_dir/tools/build/src/engine" && ./build.sh "$TOOLSET") > bootstrap.log 2>&1
++  (cd "$my_dir/tools/build/src/engine" && CXX="$CXX_FOR_BUILD" ./build.sh "$TOOLSET") > bootstrap.log 2>&1
```

Comment:
Would it not work to properly set the value of `toolset` instead?

Reply:
Probably need to set BUILD_CXX instead https://github.com/boostorg/build/blob/495b7ee9970787319dad6c70707ee8f4f18363e8/src/engine/build.sh#L41-L48

Reply:
BUILD_CXX is only used if $TOOLSET is set to cross-cxx. Moreover doing so just sets CXX to build CXX. https://github.com/boostorg/build/blob/495b7ee9970787319dad6c70707ee8f4f18363e8/src/engine/build.sh#L429

Reply:
And then `B2_CXX` is set to the value of `CXX`, so that might be how cross compiling is implemented.